README.md

```diff
@@ -2294,6 +2306,44 @@ Sinatra installs special `not_found` and `error` handlers when
 running under the development environment to display nice stack traces
 and additional debugging information in your browser.
 
+### Behavior with `raise_errors` option
+
+When `raise_errors` option is `true`, errors that are unhandled are raised 
+outside of the application. Additionally, any errors that would have been 
+caught by the catch-all error handler are raised.
+
+For example, consider the following configuration:
+
+```ruby
+# First handler
+error MyCustomError do
+  'A custom message'
+end
+
+# Second handler
+error do
+  'A catch-all message'
+end
+```
+
+If `raise_errors` is `false`:
+
+* When `MyCustomError` or descendant is raised, the first handler is invoked.
+  The HTTP response body will contain `"A custom message"` with status code 500.
+* When any other error is raised, the second handler is invoked. The HTTP 
+  response body will contain "A catch-all message" with status code 500.
+
+If `raise_errors` is `true`:
+
+* When `MyCustomError` or descendant is raised, the behavior is identical to 
+  when `raise_errors` is `false`, described above.
+* When any other error is raised, the second handler is *not* invoked, and 
+  the error is raised outside of the application.
```

Comment:
Do we need to preface this with something along the lines "if the environment is `production` or `show_exceptions` is set to `:after_handler`"?

I'm also thinking about "with status code 500", that's correct, but it can also be changed by using `status` inside the handler block, so perhaps it should be made more clear that you are not stuck with 500.

Reply:
Yes, I think there are some nuances here. I'll add that.

Reply:
> I'm also thinking about "with status code 500", that's correct, but it can also be changed by using status inside the handler block, so perhaps it should be made more clear that you are not stuck with 500.

That's true, but I guess what I was trying to describe this specific example, which isn't using the `status` method at all, so I believe it would default to 500. I didn't want to make the example too complicated, because I think this is already a bit complicated to understand.

That said, I'm happy to add some remarks about it, if you believe that it is important.

Reply:
> Do we need to preface this with something along the lines "if the environment is production or show_exceptions is set to :after_handler"?

I attempted this in 44fcd5c5, let me know what you think.

Reply:
> I attempted this in https://github.com/sinatra/sinatra/commit/44fcd5c5a9300e02420593bad808acbddf58f1c0, let me know what you think.

Looks good to me. 👍🏻

>  I guess what I was trying to describe this specific example, which isn't using the status method at all, so I believe it would default to 500. I didn't want to make the example too complicated, because I think this is already a bit complicated to understand.

Agree with that. Could the example be even simpler without mentioning the default 500 status?

I'm fine either way, these are nice improvements to the docs.

Reply:
> Could the example be even simpler without mentioning the default 500 status?

Yes, I think that is a good call. Done in 99ec9445.